### PR TITLE
Codelist/region subclass

### PR DESCRIPTION
**Pull request by luciecastella**
Related to #137 

First attempt to split the functions to have the region specific part in RegionCodeList. I renamed `_parse_dir` to `_parse_tags` because it now focuses on tags and it is more understandable. 
There are anyways loops to parse the files in `_parse_tags()` **and** in `from_dir()`, it was what seemed to be the most understandable option to me, but I can change it...

### CODE REVIEW

File: nomenclature/codelist.py [hunk 1]
```diff
@@ -221,7 +192,29 @@ def from_directory(cls, name: str, path: Path, file: str = None):
         instance of cls (CodeList if not inherited)
 
         """
-        return cls(name=name, mapping=cls._parse_dir(name, path, file))
+        code_list = []
+        # parse all files in path if file is None
+        file = file or "**/*"
```
Comment (phackstock) [inline nomenclature/codelist.py:197]: Same comment as above regarding the name of the `file` parameter. Also you could just add "**/*" as a default argument and remove this line.

**Review by phackstock — COMMENTED** [review]
Very solid start, good work @luciecastella. 
A few minor comments below.

File: nomenclature/codelist.py [hunk 2]
```diff
@@ -173,34 +167,11 @@ def _parse_dir(cls, name: str, path: Path, file: str = None) -> List[Code]:
             else:
                 # validate the schema of this codelist domain (default `generic`)
                 validate(_code_list, SCHEMA_MAPPING[cls.validation_schema])
```
[the remark this answers is not in the document]
Comment (luciecastella) [reply nomenclature/codelist.py:169]: Good point

**Comment by luciecastella** [comment]
I think that's it for this PR. Small comment: the new test to_csv() implemented by @danielhuppmann was throwing an error and I had to change `lineterminator` to `line_terminator` (older version). Might be a problem with other users with the wrong version of pandas. I don't really know how to make that better so I'll leave it to you guys.

**Review by phackstock — APPROVED** [review]
Looks all good to me. Thanks @luciecastella.

**Pull request merged**